### ISSUE TITLE
Clarify whether all tasks are intended to be executed or just those that contribute to outputs

### ISSUE THREAD

**Issue by jonrkarr** [SED-ML/sed-ml#117]
As I understand, there's a unstated understanding that only tasks that contribute to outputs are intended to be executed. @luciansmith mentioned this in another issue. This wasn't obvious to me. This is relevant for reproducibility due to things such as random number generators.

**Comment by matthiaskoenig** [comment]
I don't see this as an issue for reproducibility/replicability.
Simulations with randomness are by definition not reproducible. Every implementation will give different results (because people use different random generators, random generators change between software and library versions, and so on). So it can never be guaranteed to get the identical results between two implementations (or even different versions of one tool). As a consequence this is not an issue for reproducibility. 
It could be an issue for replicability, i.e. you try to get the same results with a single tool. But because you don't change the strategy (i.e. one day simulate all tasks, the next day only the tasks relevant for data generators) this does not matter either.

I agree we should have a clarification that "only tasks used in data generators must be executed", but otherwise I don't think this is an issue and can be closed. The strategy can be up to the tool and it does not matter for the data generator content.
In my SED-ML implementations I only execute tasks appearing in data generators.

**Comment by jonrkarr** [comment]
From a computer science perspective, SED-ML stops short of outlining a shared understanding of replicability both between and within tools. This issue is about one piece of information that's missing for such a shared understanding.

Striving for exact replicability of runs between tools is very impractical. That is not the point of this issue.

Nevertheless, I think its useful to have a common understanding of what this level of replicability means *within* each tool. This would, for example, enabled automated checking of whether each tool is replicable within itself.

**Comment by luciansmith** [comment]
I wasn't 100% sure what to do with this, but how about the following, added to the end of the opening 'Output' section (2.2.12)

"The output of a SED-ML file may be used to compare simulation executions from the same tool or from different tools.  As such, interpreters may choose to focus on the output of a SED-ML file, and execute only the tasks necessary to produce this output.  Repeated executions of the same SED-ML should always produce comparable output."

**Comment by luciansmith** [comment]
Given https://github.com/SED-ML/sed-ml/issues/115, I added my sentence from there, as well:

"The output of a SED-ML file may be used to compare simulation executions from the same tool or from different tools.  As such, interpreters may choose to focus on the output of a SED-ML file, and execute only the tasks necessary to produce this output.  Repeated executions of the same SED-ML should always produce comparable output.  When a stochastic run is given a seed, interpreters should be aware that the user expects to get identical results from repeated runs on the same architecture, even if some tasks are run in parallel."

Will this work? (@jonrkarr )

**Comment by jonrkarr** [comment]
That wording seems a bit strong. What about this (suggested change highlighted)

When a stochastic run is given a seed, interpreters should be aware that `users may expect` to get identical results from repeated runs on the same architecture, `including when` tasks are run in parallel.

**Comment by luciansmith** [comment]
Done!
